src/responder/server-nsq.ts

```diff
@@ -42,11 +47,12 @@ export class ServerNsq extends Server implements CustomTransportStrategy {
   private createConsumer(
     topic: string,
     channel: string,
-    options: any,
+    options: NsqConsumerOptions,
   ): Consumer {
-    const c = new Consumer(topic, channel, options || this.options);
-    if (this.options.discardHandler) {
-      c.reader.on('discard', this.options.discardHandler);
+    const c = new Consumer(topic, channel, options);
+
+    if (options.discardHandler) {
+      c.reader.on('discard', options.discardHandler);
```

Comment:
Need any logger here to output some information when a message is discarded?

Reply:
defer the logging to the external `discardHandler`.